src/v/storage/tests/storage_e2e_test.cc

```diff
@@ -2191,3 +2192,106 @@ FIXTURE_TEST(test_compacting_batches_of_different_types, storage_test_fixture) {
 
     BOOST_REQUIRE(before_compaction == after_compaction);
 }
+
+FIXTURE_TEST(read_write_truncate, storage_test_fixture) {
+    /**
+     * Test validating concurrent reads, writes and truncations
```

Comment:
nit: unless i'm missing something, this test doesn't have any assertion or doesn't seem to be checking for any conditions. is there some way to make it a stronger test?

Reply:
i will make this test stronger, at least checking if there is no deadlock